Signal/src/AppDelegate.m

```diff
@@ -1147,6 +1147,18 @@ - (void)checkIfAppIsReady
     [[OWSMessageReceiver sharedInstance] handleAnyUnprocessedEnvelopesAsync];
     [[OWSBatchMessageProcessor sharedInstance] handleAnyUnprocessedEnvelopesAsync];
 
+    if (!Environment.preferences.hasGeneratedThumbnails) {
+        [OWSPrimaryStorage.sharedManager.newDatabaseConnection
+            asyncReadWithBlock:^(YapDatabaseReadTransaction *_Nonnull transaction) {
```

Comment:
This is going to be disk and CPU-intensive.  Rather than use `asyncReadWithBlock`, it might be better to explicitly dispatch to a low-priority queue to avoid interfering with the UI.